code/modules/shuttle/supply.dm

```diff
@@ -596,3 +589,135 @@ GLOBAL_LIST_EMPTY(exports_types)
+	var/temp //TGUI things
+	var/last_viewed_group = "categories"
+
+/obj/item/req_tablet/examine(mob/user)
+	. = ..()
+	if(!linked_beacon)
+		to_chat(user, "There seems to be no beacon attached to this tablet, hit a beacon with a tablet to link to it.")
+	else
+		to_chat(user, "There seems to be a beacon attached to this tablet. Should be good to go for delivery.")
+	return .
+
+/obj/item/req_tablet/afterattack(atom/A, mob/user, proximity)
+	if(istype(A, /obj/item/radio/beacon))
+		to_chat(user, "You link the tablet to the beacon you hit it with, overriding the previous beacon linked if there was one.")
+		linked_beacon = A
+		playsound(get_turf(src), 'sound/machines/triple_beep.ogg', 25, TRUE)
+		icon_state = "req_tablet_on"
+
+//Begin the requisitions copypasta
+/obj/item/req_tablet/interact(mob/user)
+	. = ..()
+	if(.)
+		return
+	var/dat
+	if(temp)
+		dat = temp
+	else
+		if (SSshuttle.supply)
+			if(!linked_beacon)
+				dat = "<BR><B>WARNING: NO CONNECTIVITY WITH SUPPLY SYSTEM DUE TO LACK OF LINKED BEACON!!</B>"
+			else
+				dat += "<BR><B>Automated Launch Pod Retrieval System: Tablet edition</B>"
+				dat += "<BR><HR>Supply points: [round(SSpoints.supply_points)]<BR>"
+				dat += "<BR><HR><A href='?src=\ref[src];order=categories'>Order supply packs (CANNOT BE CANCELED ONCE ORDERED)</A>"
+				dat += "<BR><HR><A href='?src=\ref[src];send=1'>Begin drop pod delivery</A>"
+				dat += "<BR><HR>WARNING: CAN ONLY LAUNCH ITEMS TO BEACON IF THE SURROUNDING TILES OF THE BEACON HAVE A GLASS OR NO CEILING PRESENT!!"
+				dat += "<BR><B>Current approved orders: </B><BR><BR>"
+				for(var/S in SSshuttle.shoppinglist)
+					var/datum/supply_order/SO = S
+					dat += "#[SO.id] - [SO.pack.name]<BR>"
+
+	var/datum/browser/popup = new(user, "computer", "<div align='center'>Ordering Tablet</div>", 575, 450)
+	popup.set_content(dat)
+	popup.open(FALSE)
+	onclose(user, "computer")
+
+/obj/item/req_tablet/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return
+
+	if(href_list["order"])
+		if(href_list["order"] == "categories")
+			//all_supply_groups
+			//Request what?
+			last_viewed_group = "categories"
+			temp = "<b>Supply points: [round(SSpoints.supply_points)]</b><BR>"
+			temp += "<A href='?src=\ref[src];mainmenu=1'>Main Menu</A><HR><BR><BR>"
+			temp += "<b>Select a category</b><BR><BR>"
+			for(var/supply_group_name in GLOB.all_supply_groups)
+				temp += "<A href='?src=\ref[src];order=[supply_group_name]'>[supply_group_name]</A><BR>"
+		else
+			last_viewed_group = href_list["order"]
+			temp = "<b>Supply points: [round(SSpoints.supply_points)]</b><BR>"
+			temp += "<A href='?src=\ref[src];order=categories'>Back to all categories</A><HR><BR><BR>"
+			temp += "<b>Request from: [last_viewed_group]</b><BR><BR>"
+			for(var/supply_type in SSshuttle.supply_packs)
+				var/datum/supply_packs/N = SSshuttle.supply_packs[supply_type]
+				if(N.hidden || N.contraband || N.group != last_viewed_group) continue								//Have to send the type instead of a reference to
```

Comment:
```suggestion
				if(N.hidden || N.contraband || N.group != last_viewed_group)
					continue								//Have to send the type instead of a reference to
```